### PR TITLE
Use Gdk.RGBA as Gdk.Color is deprecated

### PR DESCRIPTION
**Pull request by chimosky**
@quozl kindly review.

### CODE REVIEW

**Comment by quozl** [comment]
Instantly looks wrong to me, as _Gdk.RGBA_ requires floating point inputs between 0.0 and 1.0 for each colour component, whereas _Gdk.Color_ accepts integers between 0x0 and 0xffff.

File: src/TextThought.py [hunk 1]
```diff
@@ -605,8 +606,8 @@ def _create_textview(self):
         """
 
         r, g, b = utils.gtk_to_cairo_color(self.foreground_color)
-        rgba = Gdk.Color(int(65535 * r), int(65535 * g), int(65535 * b))
-        self.textview.modify_text(Gtk.StateType.NORMAL, rgba)
+        rgba = Gdk.RGBA(int(65535 * r), int(65535 * g), int(65535 * b))
+        self.textview.modify_text(Gtk.StateType.NORMAL, rgba.to_color())
```
Comment (chimosky) [inline src/TextThought.py:610]: I'm expecting to get an error here and I'll look into it

**Comment by chimosky** [comment]
It's deprecated and we'll eventually stop using `Gdk.Color`, this is the error returned when activity is run and I try to create thoughts.

```
Traceback (most recent call last):
  File "/home/ibiam/Activities/labyrinth-activity/src/MMapArea.py", line 720, in draw
    l.draw (context)
  File "/home/ibiam/Activities/labyrinth-activity/src/Links.py", line 142, in draw
    context.set_source_rgb (self.color[0], self.color[1], self.color[2])
TypeError: 'NoneType' object has no attribute '__getitem__'
Traceback (most recent call last):
  File "/home/ibiam/Activities/labyrinth-activity/src/MMapArea.py", line 720, in draw
    l.draw (context)
  File "/home/ibiam/Activities/labyrinth-activity/src/Links.py", line 142, in draw
    context.set_source_rgb (self.color[0], self.color[1], self.color[2])
TypeError: 'NoneType' object has no attribute '__getitem__'
```
For some reason `self.color` is None when `Links.draw` is executed but `self.color` was set to `black` in L55 of `src/Links.py`.

We can also create `Gdk.RGBA` instances from `Gdk.Color` instances using `Gdk.RGBA.from_color()` which might be a fix for the error I pointed out above.

File: src/TextThought.py (hunk 1 above)
Comment (quozl) [reply src/TextThought.py:610]: Your code is in error, but I would not expect an error message.  It is another one of those situations where code can execute but the result is not what you planned.

The values are _clamped_ to the range 0.0 to 1.0, as can be seen in the [source code for Gdk.RGBA](https://github.com/GNOME/gtk/blob/bb56b4ef5d975fff91e7caa39e058d02fd7373bb/gdk/gdkrgba.c#L376).  Always look at the source code of GTK if you have any doubts as to what it does; the PyGObject documentation is often incomplete.

Use interactive testing with a Python REPL to prove what you expect;

```log
>>> from gi.repository import Gdk
>>> Gdk.RGBA(65535, 65535, 65535).to_string()
'rgb(255,255,255)'        <-- correct answer for the wrong reason due to clamping
>>> Gdk.RGBA(65535, 65535, 1).to_string()
'rgb(255,255,255)'        <-- correct answer but you may not have expected it
>>> Gdk.RGBA(65535, 65535, 0.5).to_string()
'rgb(255,255,128)'        <-- correct answer, see how BLUE is at 50% intensity
>>> Gdk.RGBA(65535, 65535, 0.5).blue
0.5        <-- can read the BLUE channel
>>> Gdk.RGBA(65535, 65535, 65535).blue
65535.0      <-- shows the clamping is happening _after_ the object is created
>>> 
```

**Comment by quozl** [comment]
> It's deprecated and we'll eventually stop using `Gdk.Color`,

I wouldn't worry.  It has been announced as deprecated for years, but it still works.

> this is the error returned when activity is run and I try to create thoughts.
> 
> ```
> Traceback (most recent call last):
>   File "/home/ibiam/Activities/labyrinth-activity/src/MMapArea.py", line 720, in draw
>     l.draw (context)
>   File "/home/ibiam/Activities/labyrinth-activity/src/Links.py", line 142, in draw
>     context.set_source_rgb (self.color[0], self.color[1], self.color[2])
> TypeError: 'NoneType' object has no attribute '__getitem__'
> Traceback (most recent call last):
>   File "/home/ibiam/Activities/labyrinth-activity/src/MMapArea.py", line 720, in draw
>     l.draw (context)
>   File "/home/ibiam/Activities/labyrinth-activity/src/Links.py", line 142, in draw
>     context.set_source_rgb (self.color[0], self.color[1], self.color[2])
> TypeError: 'NoneType' object has no attribute '__getitem__'
> ```

Sorry, can't diagnose properly, as the line numbers do not match 871646438b0155feafb604b5a335f45a8f44146d.  But it looks like it would be easy to debug.

> For some reason `self.color` is None when `Links.draw` is executed but `self.color` was set to `black` in L55 of `src/Links.py`.

Check the code executed in the order you expected; perhaps the draw callback was executed during a parent class initialisation.

> We can also create `Gdk.RGBA` instances from `Gdk.Color` instances using `Gdk.RGBA.from_color()` which might be a fix for the error I pointed out above.

I doubt it.  It should be easier to use `Gdk.RGBA.parse` instead of combining `Gdk.Color` with `Gdk.RGBA.from_color`.

**Pull request closed without merging**